x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/PostAnalysisAware.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.analysis;
+
+import org.elasticsearch.xpack.esql.common.Failures;
+import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
+
+import java.util.function.BiConsumer;
+
+public interface PostAnalysisAware {
```

Comment:
The interface needs some javadoc with examples.

Reply:
Added javadoc with examples.